lib/specinfra/ext/class.rb

```diff
@@ -2,7 +2,7 @@ class Class
   def subclasses
     result = []
     ObjectSpace.each_object(Class) do |k|
-      next if k.name.nil?
+      next unless k == k.ancestors.first #Skips if this is a singleton class
```

Comment:
The [`singleton_class`](http://ruby-doc.org/core/Object.html#method-i-singleton_class) I am skipping is not the same as a [`Singleton`](http://ruby-doc.org/stdlib/libdoc/singleton/rdoc/Singleton.html). The similar naming is confusing. The `singleton_class` is a concept ruby uses to add methods to an object without affecting other objects of the same class. However, the first ancestor is still equal to itself so this isn't a workable solution.

```ruby
class Foo
end

s = Foo.singleton_class
s.ancestors.first == s
```

Gives this result:

```ruby
true
```